routes/routes.go

```diff
@@ -84,11 +83,28 @@ func Init(ctx context.Context, r *mux.Router, cfg *config.Config, clients Client
 
 	r.StrictSlash(true).Path("/filters/{filterID}/use-latest-version").HandlerFunc(filter.UseLatest)
 
-	// Create debug endpoint, if required to do so, under auth middleware
+	// Enable profiling endpoint for authorised users
 	if cfg.EnableProfiler {
-		idCli := identity.NewAPIClient(clients.ZebedeeHealth.Client, cfg.ZebedeeURL)
-		identityHandler := identityMiddleware.HandlerForHTTPClient(idCli)
-		middlewareChain := alice.New(identityHandler).Then(http.DefaultServeMux)
-		r.PathPrefix("/debug/").Handler(middlewareChain)
+		middlewareChain := alice.New(profileMiddleware(cfg.PprofToken)).Then(http.DefaultServeMux)
+		r.PathPrefix("/debug").Handler(middlewareChain)
+	}
+}
+
+// profileMiddleware to validate auth token before accessing endpoint
+func profileMiddleware(token string) func(http.Handler) http.Handler {
+	return func(h http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			ctx := req.Context()
+
+			pprofToken := req.Header.Get("Authorization")
+			if pprofToken == "Bearer " || pprofToken != "Bearer "+token {
+				log.Event(ctx, "invalid auth token", log.ERROR, log.Error(errors.New("invalid auth token")))
+				w.WriteHeader(404)
+				return
+			}
+
+			log.Event(ctx, "accessing profiling endpoint", log.INFO)
+			h.ServeHTTP(w, req)
+		})
```

Comment:
There is nothing here that is specific to this API. Is this likely to be done in other apps? If so I wonder if there is value in moving it to a common library rather than duplicating it every time.

Reply:
Only the two applications at the moment - not sure which library I should add it to or if we would end up creating a dp-profiling-go library?

Reply:
Possibly move it to dp-net?

Reply:
Apart from the log message and variable names, there doesn't seem to be anything that is specific to pprof. I wonder if we could make it a generic "token check" middleware if it's moved out?

Reply:
I think if this is rolled out to future applications, then we should add a story to move this to a library for re-use, for now I will merge.